src/Extensions/TimedScopes/Internal/DiagnosticsObserversInitializer.cs

```diff
@@ -144,9 +149,46 @@ private void OnActivityStopped(Activity activity, object payload)
 			}
 		}
 
-		private void OnException(string eventName, object payload)
+		private void OnException(string eventName, object payload) =>
+			m_logger.LogError(
+				Tag.Create(),
+				ExtractExceptionFromPayload(payload),
+				"Exception diagnostic event '{0}' with payload '{1}'", eventName, payload);
+
+
+		// Made internal for unit testing
+		internal static Exception? ExtractExceptionFromPayload(object? payload)
 		{
-			m_logger.LogError(Tag.Create(), payload as Exception, "Exception diagnostic event '{0}'", eventName);
+			if (payload == null)
+			{
+				return null;
+			}
+			else if (payload is Exception exception)
+			{
+				return exception;
+			}
+			else
+			{
+				// Attempting to find exception property since payload often use private classes,
+				// It would be completely removed after .net 5 release.
```

Comment:
>net [](start = 45, length = 3)

NET

Reply:
Thank you, I'll change it in a future PR's